vllm/core/block/interfaces.py

```diff
@@ -90,21 +109,33 @@ def get_common_computed_block_ids(
             self, seq_block_ids: List[List[int]]) -> List[int]:
         pass
 
+    @abstractmethod
+    def cow_block_if_not_appendable(self, block: Block) -> Optional["BlockId"]:
+        pass
+
+    @abstractmethod
+    def promote_to_immutable_block(self, block: Block) -> BlockId:
+        pass
+
```

Comment:
This makes the `BlockAllocator` abstraction leaky; users of the class should never know about these methods (only `NaiveBlock` knows about `cow_block_if_not_appendable`, and `PrefixCachingBlock` only knows about `promote_to_immutable_block`).

Reply:
My intuition says python typing is not developed enough to easily represent this (lmk if I'm wrong). At the very least we should add comments explaining these shouldn't be used by users, only the implementation itself..

Reply:
This is not working actually because of this line https://github.com/vllm-project/vllm/blob/d6f4bd7cddc9546c38568c92c3772d22940a09f2/vllm/core/block/prefix_caching_block.py#L360

So, based on what you said, it assumes NaiveBlock can only accept NavieBlockAllocator, but it is currently not true.

Reply:
So, I think if we want to make this part fixed, we should change this code. I will probably leave it up to you after this PR?

Reply:
I see -- can we add some comments that these methods shouldn't be used by anyone besides block allocators?